File: src/CadastroIgreja.Domain/Entities/CartaPregacao.cs
namespace CadastroIgreja.Domain.Entities;

public class CartaPregacao
{
    public Guid Id { get; set; } = Guid.NewGuid();
    public string Numero { get; set; } = string.Empty;
    public Guid UsuarioId { get; set; }
    public Usuario? Usuario { get; set; }
    public Guid SolicitacaoPregadorId { get; set; }
    public SolicitacaoPregador? SolicitacaoPregador { get; set; }
    public DateTime DataEmissao { get; set; } = DateTime.UtcNow;
    public DateTime DataValidade { get; set; } = DateTime.UtcNow.AddYears(1);
    public string QrCode { get; set; } = string.Empty;
    public string? PdfPath { get; set; }
    public bool Ativa { get; set; } = true;
}

File: frontend/lib/src/models/carta_pregacao.dart
class CartaPregacao {
  const CartaPregacao({required this.id, required this.numero, required this.usuarioId, required this.ativa});

  final String id;
  final String numero;
  final String usuarioId;
  final bool ativa;

  factory CartaPregacao.fromJson(Map<String, Object?> json) {
    return CartaPregacao(
      id: json['id'] as String? ?? '',
      numero: json['numero'] as String? ?? '',
      usuarioId: json['usuarioId'] as String? ?? '',
      ativa: json['ativa'] as bool? ?? false,
    );
  }
}
